src/components/menu/ui/getLinkComponent.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { Fragment, ReactElement, cloneElement, createElement, useMemo } from "react";
+import { isExternalLink } from "../../../shared/utils/isExternalLink";
+
+type LinkComponentProps = {
+  children?: React.ReactNode;
+  className?: string;
+};
+
+export const getLinkComponent = (linkComponent?: ReactElement, href?: string) => {
```

Comment:
linkComponent would be better a function like `(href, className, children) => ReactNode`, instead of element. Now we need to make something like `{linkComponent: <MyLink href={href} />}` and we could add here only a className. We should expose our current link component and use simple wrapper `linkComponent: (href, className, children) => {const cb = useCallback(ReactRouter); return <OurLinkComponent onClick(cb) href={href} className={className}/>`

Reply:
Changed it to render function.